### PR TITLE
Add rollup import transform

### PR DESCRIPTION
**Pull request by marklundin**
This PR addresses a long standing issue with the UMD build not running on devices without dynamic `import()` support. See #6011.

This issue stemmed from the use of dynamic `import()` statement, where `import` is a reserved keyword and would throw a SyntaxError during parsing on older browsers, even if the code path was never executed.

This PR adds a rollup transform to the UMD build only, which wraps `import()` statements in a dynamic function. This is only evaluated at runtime, thereby avoiding the parsing on load error. Note that this does not polyfill dynamic imports, it will simply prevent parsing errors on browser without support.

### Caveats
This does put a limitation on CSP for UMD builds, however this only applies if *all of the following conditions are met*

1. The UMD build is used (not applicable to the ES6 build) **AND**
2. Hosting blocks `unsafe-eval` in their CSP **AND**
3. When using either WebGPU or importing an ESM ScriptType

I confirm I have read the [contributing guidelines](https://github.com/playcanvas/engine/blob/master/.github/CONTRIBUTING.md) and signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

File: utils/rollup-dynamic-import-transform.mjs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+/**
+ * This rollup plugin transform code with dynamic import statements and wraps them
+ * in a `new Function('import("modulePath")')` statement, in order to avoid parsing errors in older browsers
+ * without support for dynamic imports.
+ *
+ * Note that whilst this will prevent parsing errors, it will
```
Comment (mvaligursky) [inline utils/rollup-dynamic-import-transform.mjs:6]: hmm looks half finished sentence? :)

**Review by mvaligursky — APPROVED** [review]
Looks great, but please give some time for additional opinions before merging.

@kungfooman perhaps?

**Review by kungfooman — APPROVED** [review]
Looks very good 👍

File: utils/rollup-dynamic-import-transform.mjs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        name: 'dynamic-import-transform',
+        transform(code, id) {
+            /**
+             * Transforms the code by replacing `import(` with `new Function("return import")(`.
+             * @param {string} code - The code to transform.
+             * @param {string} id - The id of the code.
+             * @returns {object} - The transformed code and map.
+             */
+            return {
+                code: code.replace(/import\(/g, 'new Function("modulePath", "return import(modulePath)")('),
```
[the remark this answers is not in the document]
Comment (kpal81xd) [reply utils/rollup-dynamic-import-transform.mjs:20]: @marklundin  Just change the regex to be `/[^\w]import/g` which unmatch any character in front of `import`

Comment (kpal81xd) [inline utils/rollup-dynamic-import-transform.mjs:20]:
```suggestion
                code: code.replace(/([\w^])import\(/g, '$1new Function("modulePath", "return import(modulePath)")('),
```

**Review by kpal81xd — APPROVED** [review]

**Review by slimbuck — APPROVED** [review]

**Pull request merged**